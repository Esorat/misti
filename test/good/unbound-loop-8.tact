fun test() {
    let i: Int = 0;
    while (10 < i) {} // warning: unbound loop
}

