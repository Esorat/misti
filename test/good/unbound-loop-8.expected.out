[HIGH] UnboundLoops: Found unbounded loop
test/good/unbound-loop-8.tact:3:17:
  2 |     let i: Int = 0;
> 3 |     while (10 < i) {} // warning: unbound loop
                      ^
  4 | }
The condition variable doesn't change inside the loop
Help: Consider changing the variable inside the loop to ensure it terminates
See: https://nowarp.io/tools/misti/docs/detectors/UnboundLoops

[MEDIUM] ReadOnlyVariables: Read-only variable
test/good/unbound-loop-8.tact:2:9:
  1 | fun test() {
> 2 |     let i: Int = 0;
              ^
  3 |     while (10 < i) {} // warning: unbound loop
Help: Consider creating a constant instead
See: https://nowarp.io/tools/misti/docs/detectors/ReadOnlyVariables